resourcematch-api/main.py

```diff
@@ -62,8 +64,16 @@ class MatchRequest(BaseModel):
     text: str
 
 
+class SearchResult(BaseModel):
+    url: AnyHttpUrl
+    book_title: str
+    chapter_title: str
+    page_title: str
+    visible_content: str
+
+
 class MatchResponse(BaseModel):
-    urls: List[AnyHttpUrl]
+    search_results: List[SearchResult]
```

Comment:
Minor / design choice: It may be that you explicitly want to nest the results as a field (e.g. in case we want to add other fields). However, just in case, it is possible to avoid this and return an array of objects something like:

```python
class MatchResponse(BaseModel):
    url: AnyHttpUrl
    book_title: str
    chapter_title: str
    page_title: str
    visible_content: str
```

You could then change the signature of `perform_match` accordingly:

```python
@app.post("/match")
async def perform_match(match_request: MatchRequest) -> List[MatchResponse]:
```